Source/WebCore/layout/formattingContexts/inline/InlineContentConstrainer.cpp

```diff
@@ -174,19 +174,24 @@ void InlineContentConstrainer::updateCachedWidths()
         auto isWordSeparator = false;
         if (auto* textItem = dynamicDowncast<InlineTextItem>(item))
             isWordSeparator = textItem->isWordSeparator();
-
-        m_inlineItemWidths[i] = m_inlineFormattingContext.formattingUtils().inlineItemWidth(item, 0, false) +  (isWordSeparator ? item.style().wordSpacing() : 0.0f);
-        m_inlineItemWidthsMax = std::max(m_inlineItemWidthsMax, m_inlineItemWidths[i]);
-        m_firstLineStyleInlineItemWidths[i] = m_inlineFormattingContext.formattingUtils().inlineItemWidth(item, 0, true) + (isWordSeparator ? item.firstLineStyle().wordSpacing() : 0.0f);
-        m_inlineItemWidthsMax = std::max(m_inlineItemWidthsMax, m_firstLineStyleInlineItemWidths[i]);
+        // Opaque items are ignored by inline layout. Skip over this item by setting its width to 0.
+        if (item.isOpaque()) {
+            m_inlineItemWidths[i] = 0;
+            m_firstLineStyleInlineItemWidths[i] = 0;
```

Comment:
aren't these 0.f by default? meaning that maybe all you need to do is `if (!item.isOpaque())`